packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -218,8 +212,15 @@ class StorageS3Service {
     );
     final keyToGetUrl = '$resolvedPrefix$key';
 
-    final host =
-        '$_defaultBucket.${_getS3EndpointHost(region: _defaultRegion)}';
+    var host = '$_defaultBucket.${_getS3EndpointHost(region: _defaultRegion)}';
+
+    if (options.useAcceleration) {
+      // https: //docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration-getting-started.html
+      host = host
+          .replaceFirst(RegExp('$_defaultRegion\\.'), '')
+          .replaceFirst(RegExp(r'\.s3\.'), '.s3-accelerate.');
```

Comment:
The logic is from here:

https://github.com/aws-amplify/amplify-flutter/blob/fc6736512e7fbd595a3b376f3d7f2b3199e47d06/packages/storage/amplify_storage_s3_dart/lib/src/sdk/src/s3/operation/get_object_operation.dart#L393-L396

Could make this logic a util function in the sdk and export it to use here?